### ISSUE TITLE
Invalid "is defined but never used" errors

### ISSUE THREAD

**Issue by sheerun** [saadq/lynt#37]
Example file:

```es6
import React from 'react'

class Hero extends React.PureComponent {
  render () {
    return (
      <section className='hero is-large has-bg-gray-10'>
      </section>
    )
  }
}

class Brokers extends React.PureComponent {
  render () {
    return (
      <div className='page brokers'>
        <Hero />
      </div>
    )
  }
}

export default Brokers
```

Lyft returns:

```
  3:7  'Hero' is defined but never used.  no-unused-vars
```

**Comment by saadq** [comment]
By default, lynt only acts as a minimal, vanilla JS linter. It has builtin "modes" you can specify with flags which will add additiona. For example, if you're using TypeScript you can use `--typescript`, with React you can use `--react`, etc and it will add in a bunch of additional react rules/support.

So to get this to work, you should be using `lynt --react` instead of just `lynt`.

However, ESLint 5 introduced a breaking change to how linting works, and you cannot pass invalid globs to it – otherwise it will throw a fatal error. By default, if no globs are passed to Lynt I was passing `['**/*.js', '**/*.jsx']` for you, but if you don't have any .jsx files it will now fail.

The current way to use lint with React is to do the following:

```
lynt src --react
```

where `src` can be any glob, but a glob does need to currently be passed to prevent errors. The next release I push will fix this new ESLint change so that `lynt --react` will be enough. 

After that, I can see if adding `--react` by default would be a good idea.

**Comment by saadq** [comment]
The issue I mentioned above is going to be tracked in #39

**Comment by sheerun** [comment]
I'm for convention over configuration :) If you don't want to enable react rules by default maybe auto-detect react by reading package.json?

**Comment by saadq** [comment]
I've been thinking about this, and I think there is no harm in just always allowing react rules/support. But I can conditionally add flow support if the user contains a `.flowconfig` file and conditionally add typescript support if the user has a `tsconfig.json` file. 

So the necessity to use `--react`, or `--flow` or `--typescript` flags would be removed. I think I'll be going with this path,

**Comment by saadq** [comment]
#39 has been resolved as of v0.5.5.

You just need to run `lynt --react` and you don't need to pass any file globs for it to lint your project.

#40 is next on the board where I will add support to automatically detect if the user needs React/TypeScript/Flow support (so `--react`, `--typescript` or `--flow` flags will no longer need to be passed in).

I can't add React support by default, because ESLint throws some warnings when React support is added without the user having React installed. So I'll be going with your suggestion to check whether the user needs React support by reading `package.json`.

**Comment by sheerun** [comment]
great, thank you :)
